pkg/detectors/meraki/meraki.go

```diff
@@ -0,0 +1,115 @@
+package meraki
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	apiKey = regexp.MustCompile(detectors.PrefixRegex([]string{"meraki"}) + `([0-9a-f]{40})`)
+)
+
+func (s Scanner) getClient() *http.Client {
+	if s.client != nil {
+		return s.client
+	}
+
+	return defaultClient
+}
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"meraki"}
+}
+
+func (s Scanner) Description() string {
+	return "Cisco Meraki is a cloud-managed IT solution that provides networking, security, and device management through an easy-to-use interface." +
+		"Meraki APIs make it possible to rapidly deploy and manage networks at scale, build on a platform of intelligent, cloud-connected IT products, and engage with users in powerful new ways."
+}
+
+// FromData will find and optionally verify Meraki API Key secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	// uniqueMatches will hold unique match values and ensure we only process unique matches found in the data string
+	var uniqueMatches = make(map[string]bool)
```

Comment:
Not really a big deal, but for sets the convention is to use `map[string]struct{}` instead of `map[string]bool`. This is both because the value of `bool` isn't used and empty structs save allocations.

e.g., https://www.reddit.com/r/golang/comments/15vaxlu/the_set_data_structure_in_go/
```suggestion
	var uniqueMatches = make(map[string]struct{})
```